github/store/db.go

```diff
@@ -47,72 +47,180 @@ const (
+	"github_pull_request_reviews_versioned",
+	"github_pull_request_comments_versioned",
 }
 
 func (s *DB) SetActiveVersion(ctx context.Context, v int) error {
-	// TODO: for some reason the normal parameter interpolation $1 fails with
-	// pq: got 1 parameters but the statement requires 0
+	// Unified schema
 
-	_, err := s.DB.ExecContext(ctx, fmt.Sprintf(`CREATE OR REPLACE VIEW organizations AS
+	_, err := s.DB.ExecContext(ctx, "DROP MATERIALIZED VIEW IF EXISTS owners")
+	if err != nil {
+		return fmt.Errorf("failed to drop unified view owners: %v", err)
+	}
+	_, err = s.DB.ExecContext(ctx, fmt.Sprintf(`CREATE MATERIALIZED VIEW owners AS
+	SELECT login, name
+	FROM github_organizations_versioned WHERE %v = ANY(versions)`, v))
+	if err != nil {
+		return fmt.Errorf("failed to create unified view owners: %v", err)
+	}
+
+	_, err = s.DB.ExecContext(ctx, "DROP MATERIALIZED VIEW IF EXISTS users")
+	if err != nil {
+		return fmt.Errorf("failed to drop unified view users: %v", err)
+	}
+	_, err = s.DB.ExecContext(ctx, fmt.Sprintf(`CREATE MATERIALIZED VIEW users AS
+	SELECT login, name
+	FROM github_users_versioned WHERE %v = ANY(versions)`, v))
+	if err != nil {
+		return fmt.Errorf("failed to create unified view users: %v", err)
+	}
+
+	_, err = s.DB.ExecContext(ctx, "DROP MATERIALIZED VIEW IF EXISTS repositories")
+	if err != nil {
+		return fmt.Errorf("failed to drop unified view repositories: %v", err)
+	}
+	_, err = s.DB.ExecContext(ctx, fmt.Sprintf(`CREATE MATERIALIZED VIEW repositories AS
+	SELECT owner_login AS owner, name, full_name, private, description
+	FROM github_repositories_versioned WHERE %v = ANY(versions)`, v))
+	if err != nil {
+		return fmt.Errorf("failed to create unified view repositories: %v", err)
+	}
+
+	_, err = s.DB.ExecContext(ctx, "DROP MATERIALIZED VIEW IF EXISTS issues")
+	if err != nil {
+		return fmt.Errorf("failed to drop unified view issues: %v", err)
+	}
+	_, err = s.DB.ExecContext(ctx, fmt.Sprintf(`CREATE MATERIALIZED VIEW issues AS
+	SELECT repository_owner, repository_name, repository_owner || '/' || repository_name AS repository_full_name,
+           number, state, title, body,
+           created_at, closed_at, updated_at, comments, user_id, user_login, htmlurl AS html_url, labels
+	FROM github_issues_versioned WHERE %v = ANY(versions)`, v))
+	if err != nil {
+		return fmt.Errorf("failed to create unified view issues: %v", err)
+	}
+
+	_, err = s.DB.ExecContext(ctx, "DROP MATERIALIZED VIEW IF EXISTS issue_comments")
+	if err != nil {
+		return fmt.Errorf("failed to drop unified view issue_comments: %v", err)
+	}
+	_, err = s.DB.ExecContext(ctx, fmt.Sprintf(`CREATE MATERIALIZED VIEW issue_comments AS
+	SELECT c.repository_owner, c.repository_name, c.repository_owner || '/' || c.repository_name AS repository_full_name,
+	       c.issue_number, c.created_at, c.body, c.user_id, c.user_login, c.htmlurl AS html_url
+	FROM github_issue_comments_versioned AS c
+	JOIN github_issues_versioned AS i ON
```

Comment:
why can't we use only the content of `github_issue_comments_versioned`?

Reply:
`github_issue_comments_versioned` contains both comments for PRs and Issues.
In unified schema `issue_comments` contains only comments for issues and `pull_request_comments` only for pull requests. To distinguish them we need this join.